### PR TITLE
Feat gh proj 2 heartbeat calculation improvement

### PR DESCRIPTION
**Pull request by JaydenLiang**
Incorporate the autoscale-core changes made in: https://github.com/fortinet/autoscale-core/projects/3

### CODE REVIEW

**Comment by JaydenLiang** [comment]
latest QA/DEV release created: [3.5.0-dev.0](https://github.com/fortinet/fortigate-autoscale-aws/releases/tag/3.5.0-dev.0)

File: test/fortigate-heartbeat-sync.spec.ts [hunk 1]
```diff
@@ -35,12 +36,24 @@ describe('FortiGate first heartbeat sync.', () => {
         awsTestMan = new AwsTestMan(mockDataRootDir);
     });
     after(function() {
-        mockEC2.restoreAll();
-        mockS3.restoreAll();
-        mockElbv2.restoreAll();
-        mockLambda.restoreAll();
-        mockAutoscaling.restoreAll();
-        mocDocClient.restoreAll();
+        if (mockEC2) {
```
Comment (jamie-pate) [inline test/fortigate-heartbeat-sync.spec.ts:39]: could use `mockEC2?.restoreAll()` etc

**Review by jamie-pate — APPROVED** [review]

**Review by Joel-Cripps — APPROVED** [review]

**Comment by JaydenLiang** [comment]
latest QA/DEV release created: [3.5.0-dev.1](https://github.com/fortinet/fortigate-autoscale-aws/releases/tag/3.5.0-dev.1)

**Comment by JaydenLiang** [comment]
@fshuva 3.5.0-dev.0 is the same as 3.5.0.dev.1 in terms of function codes and functionalities.

**Comment by JaydenLiang** [comment]
latest QA/DEV release created: [3.5.0-dev.2](https://github.com/fortinet/fortigate-autoscale-aws/releases/tag/3.5.0-dev.2)

**Comment by JaydenLiang** [comment]
latest QA/DEV release created: [3.5.0-dev.3](https://github.com/fortinet/fortigate-autoscale-aws/releases/tag/3.5.0-dev.3)

**Pull request merged**